cps/main.c

```diff
@@ -1026,12 +1029,14 @@ cps_stage2(void *arg)
 }
 
 int
-run_cps(struct net_config *net_conf, struct cps_config *cps_conf,
+run_cps(struct net_config *net_conf,
+	struct gk_config *gk_conf, struct cps_config *cps_conf,
 	struct lls_config *lls_conf, const char *kni_kmod_path)
 {
 	int ret;
 
-	if (net_conf == NULL || cps_conf == NULL || lls_conf == NULL) {
+	if (net_conf == NULL || gk_conf == NULL ||
+			cps_conf == NULL || lls_conf == NULL) {
```

Comment:
Bug: on a Grantor server, `gk_conf == NULL`.

The role of the CPS block in a Grantor server is to establish the ECMP protocol between the grantor server and the gateway/router in order to load balance between Grantor servers.

In a Grantor server, all rtnetlink requests should only produce warnings in the log.

You'll need to make a number of small adjustments in the code to fix this bug.